sldns/wire2str.c

```diff
@@ -1139,11 +1141,37 @@ static int sldns_wire2str_svcparam_ech2str(char** s,
 	(*s) += size;
 	(*slen) -= size;
 
-	w += sldns_str_print(s, slen, "\"");	
+	w += sldns_str_print(s, slen, "\"");
 
 	return w + size;
 }
 
+static int sldns_wire2str_svcparam_dohpath2str(char** s,
+	size_t* slen, uint16_t data_len, uint8_t* data)
+{
+	int w = 0;
+	uint16_t i;
+
+	assert(data_len > 0); /* Guaranteed by sldns_wire2str_svcparam_scan */
+
+	w += sldns_str_print(s, slen, "=\"");
+
+	/* RC6570#section-2.1 specifies that the '\' (and other non-letter
+	 * characters in the URI) are "intended to be copied literally" (as 
+	 * opposed to the alpn printing) */
+	for (i = 0; i < data_len; i++) {
+		if (!isprint(data[i])) {
+			w += sldns_str_print(s, slen, "\\%03u", (unsigned) data[i]);
+		} else {
+			w += sldns_str_print(s, slen, "%c", data[i]);
+		}
+	}
+
+	w += sldns_str_print(s, slen, "\"");
+
+	return w;
+}
+
```

Comment:
```suggestion
```
Escaped quotes and escaped backslashes are not printed correctly, but they can be in the input.
At least what you print should result into the same as what you read.
Together with the suggestion below, this can just be removed.